### ISSUE TITLE
[Good First Issue]: Align behavior of ONNX Frontend operator ReduceProd-11, 13, 18 with original framework

### ISSUE THREAD

**Issue by gkrivor** [openvinotoolkit/openvino#20558]
### Context

Neural networks are graphs consisting of nodes called operators. Each operator corresponds to a mathematical function, usually described in framework's documentation or an AI standard, such as ONNX.
OpenVINO ONNX Frontend is a component responsible for working with ONNX graphs and requires implementation of different ONNX operators in order to use ONNX models.
This task requires alignment between OpenVINO ONNX Frontend and original framework implementations of ReduceProd for next list of opsets: [opset 11](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceProd-11),  [opset 13](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceProd-13),  [opset 18](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceProd-18)
Necessary help will be provided by ONNX Fronted team.

### What needs to be done?

First of all, please, take a look on [ReduceMax](https://github.com/openvinotoolkit/openvino/pull/23475) PR for a reference.

Operator details can be found in [ONNX Operators](https://github.com/onnx/onnx/blob/main/docs/Operators.md#ReduceProd)
More details can be found in ONNX Changelog: [opset 11](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceProd-11),  [opset 13](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceProd-13),  [opset 18](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceProd-18)
1. Operator already has a common implementation in OpenVINO. First of all, you need to review a documentation and prepare a table with differences between versions. It could be, for instance, a missing property, extended/reduced coverage of existing property, etc...
2. Copy existing implementation [here](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/onnx/frontend/src/op) to make it aligned with original framework (extend or reduce coverage of a common implementation). Copy of modified implementation should be in a defined opset, or in opset 1 in case it implements oldest implementation. [Example of multi-opset operation](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/op/softmax.cpp#L50).
3. Register the function in [ops_bridge.cpp](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/ops_bridge.cpp#L554-L560) while keeping alphabetical order
4. Create test model(s) in [ONNX models directory](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/onnx/tests/models/). OpenVINO test infrastructure then converts prototxt files to ONNX models - you will use those models later in tests
5. Add tests covering all use cases [here](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/tests/onnx_import.in.cpp)
6. Check [Python xfailed tests](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/tests/tests_python/test_backend.py) to find a test marked as a xfailed for added functionality. If any exist - remove corresponding lines and try to verify by using cmdline "python -m pytest -k name_of_test".
More details in [adding operators to ONNX Frontend guide](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/how_to_add_op.md)

### Resources

- [Contribution guide - start here!](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [Intel DevHub Discord channel](https://discord.gg/wPuqAujS) - engage in discussions, ask questions and talk to OpenVINO developers
- [What is OpenVINO?](https://github.com/openvinotoolkit/openvino#what-is-openvino-toolkit)
- [User documentation](https://docs.openvino.ai/)
- Guides on [ONNX Frontend architecture](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/architecture.md) and [ONNX Frontend tests](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/tests.md) are also available
- You can also see a [blog post on contributing to OpenVINO](https://medium.com/openvino-toolkit/how-to-contribute-to-an-ai-open-source-project-c741f48e009e)

### Contact points

@gkrivor

**Comment by PulkitGarg777** [comment]
.take

**Comment by p-wysocki** [comment]
Hi @PulkitGarg777, can we help you somehow?

**Comment by PulkitGarg777** [comment]
Hi @p-wysocki, sorry for not looking into it, been kinda busy. I've seen similar issues. I'll study them and if i get stuck somewhere, I'll surely let you know. Thank you for reaching out `:)`

**Comment by p-wysocki** [comment]
Hi @PulkitGarg777, are you still on that issue or can we reassign it to other contributors?

**Comment by PulkitGarg777** [comment]
You can reassign this issue, been busy with other work, my apologies

**Comment by mlukasze** [comment]
No worries, check us out in more convenient moment for you, we will wait :)
I will reassign this ticket, but there will be more to train on.

**Comment by abhilashw** [comment]
.take

**Comment by abhilashw** [comment]
Below are the differences between opset1 and opset11, opset13, opset18
![image](https://github.com/openvinotoolkit/openvino/assets/32544781/a1493db3-fbf5-4b37-85bc-57552ebd6401)

**Comment by mlukasze** [comment]
@gkrivor take a look, please

**Comment by p-wysocki** [comment]
cc @gkrivor

**Comment by inbasperu** [comment]
Hi @abhilashw, just wanted to check if you're still working on this issue. If not, I'd be keen to take it on if it's available! 
Thanks

**Comment by andrei-kochin** [comment]
@abhilashw so you have identified the difference
Are you going to implement the code changes to have a generic implementation or do you need any additional feedback?

**Comment by abhilashw** [comment]
@andrei-kochin I was waiting for @gkrivor  to provide feedback.
Yeah I have made some changes to code. Working on adding testcases

**Comment by gkrivor** [comment]
@abhilashw hi, do you have any progress or I can take this task and complete?

**Comment by gkrivor** [comment]
.take

**Comment by hub-bla** [comment]
@mlukasze, @gkrivor, this one also #25875

**Comment by geeky33** [comment]
.take

**Comment by p-wysocki** [comment]
I'm unassigning the task due to assignee's inactivity.

**Comment by thesakshidiggikar** [comment]
.take

**Comment by thesakshidiggikar** [comment]
@gkrivor  hi need your help i did changes which and tried to solve the issue bu i try to build do  test run it gives me error

**Comment by p-wysocki** [comment]
OpenVINO is a large project and longer build times are expected, especially on older machines. In order to reduce the build time you may disable unneeded components with CMake flags described [here](https://github.com/openvinotoolkit/openvino/blob/master/docs/dev/cmake_options_for_custom_compilation.md).

**Comment by p-wysocki** [comment]
I'm unassigning the issue due to current assignee's inactivity.

**Comment by adisan-dev** [comment]
.take

**Comment by adisan-dev** [comment]
Hi @p-wysocki @gkrivor @mlukasze ,
I just realised, by any chance, is this issue already solved [here](https://github.com/openvinotoolkit/openvino/pull/25875) and was left open accidentally?

**Comment by p-wysocki** [comment]
It seems so. Sorry for the confusion!
